### ISSUE TITLE
Incorrect GIT_WORK_DIR with fish

### ISSUE THREAD

**Issue by jonathf** [TheLocehiliosan/yadm#345]
### Describe the bug

yadm lists all my tracked files as deleted when running `yadm status` in any folder except $HOME.

### To reproduce

Can this be reproduced with the yadm/testbed docker image: No

testbed doesn't allow easy install of fish.

1. # Install fish and yadm
2. yadm clone <any/repo>
3. fish
4. cd ~/.config
5. yadm status

### Expected behavior

Status should list current status of your dotfiles.
Instead it lists all dotfiles as deleted, as if the `GIT_WORK_TREE=.` is current directory instead of `$HOME`.
In other words, if you are in your $HOME folder, things works as expected.

### Environment

 - Operating system: Arch Linux
 - Version yadm: 3.1.0
 - Version Git: 2.31.1
 - Fish version: 3.2.2

**Comment by tmewett** [comment]
To help bug-hunting: I **cannot** reproduce this with

* OS: Debian 10
* yadm 3.1.0
* git 2.20.1
* fish 3.0.2

**Comment by jonathf** [comment]
I have a laptop with similar specs with Ubuntu. Also no issue.

Note that expliclitly setting `$GIT_WORK_TREE` also works. So for the time being, as a workaround, I have the following alias in my `config.fish`:
```fish
alias yadm="env GIT_WORK_TREE=$HOME yadm"
```

**Comment by erijo** [comment]
Based on [this comment](https://github.com/yadm-dev/yadm/pull/333#issuecomment-1319847191) I'm closing this issue assuming it was caused by a bug that happened when clone wasn't run in $HOME (which is now fixed).
